### PR TITLE
Add remaining sounds

### PR DESCRIPTION
**Pull request by AIOOB**
I found a few sounds that had been left out the API.

### CODE REVIEW

**Comment by simon816** [comment]
While you're at it, could you fix a slight inconsistency with naming that I noticed?
`SILVERFISH_KILL` should be `SILVERFISH_DEATH` to match with all other mob death sounds.

File: src/main/java/org/spongepowered/api/effect/sound/SoundTypes.java [hunk 1]
```diff
@@ -138,11 +170,19 @@
     public static final SoundType MAGMACUBE_WALK = null;
     public static final SoundType MAGMACUBE_WALK2 = null;
     public static final SoundType MAGMACUBE_JUMP = null;
+    public static final SoundType NEUTRAL_DIE = null;
```
Comment (modwizcode) [inline src/main/java/org/spongepowered/api/effect/sound/SoundTypes.java]: Should be NEUTRAL_DEATH?

File: src/main/java/org/spongepowered/api/effect/sound/SoundTypes.java [hunk 2]
```diff
@@ -138,11 +170,19 @@
...
+    public static final SoundType NEUTRAL_HURT = null;
+    public static final SoundType NEUTRAL_FALL_BIG = null;
+    public static final SoundType NEUTRAL_FALL_SMALL = null;
+    public static final SoundType NEUTRAL_SWIM = null;
+    public static final SoundType NEUTRAL_SPLASH = null;
     public static final SoundType PIG_IDLE = null;
     public static final SoundType PIG_DEATH = null;
     public static final SoundType PIG_WALK = null;
+    public static final SoundType PLAYER_DIE = null;
```
Comment (modwizcode) [inline src/main/java/org/spongepowered/api/effect/sound/SoundTypes.java]: Should be PLAYER_DEATH

File: src/main/java/org/spongepowered/api/effect/sound/SoundTypes.java [hunk 3]
```diff
@@ -131,6 +157,12 @@
     public static final SoundType GUARDIAN_DEATH = null;
     public static final SoundType GUARDIAN_ELDER_DEATH = null;
     public static final SoundType GUARDIAN_LAND_DEATH = null;
+    public static final SoundType HOSTILE_DIE = null;
```
Comment (modwizcode) [inline src/main/java/org/spongepowered/api/effect/sound/SoundTypes.java]: Should be HOSTILE_DEATH

**Comment by me4502** [comment]
@modwizcode Looking good to merge? A part of the implementation is waiting on this.

**Comment by modwizcode** [comment]
Looks good to me, I'll merge when I'm back home.

**Comment by Zidane** [comment]
DO NOT HIT THE MERGE BUTTON.

Do it right and cherry-pick it into master.

**Comment by ST-DDT** [comment]
@Zidane Why? I haven't seen any attached commit that would cause merge conflicts.

**Comment by modwizcode** [comment]
That's why I gotta get to a PC.

**Comment by Mumfrey** [comment]
> @Zidane Why? I haven't seen any attached commit that would cause merge conflicts.

Clerical rather than technical, GitHub automatic merges are ugly as hell in the commit log.

**Pull request closed without merging**